src/providers/idp/idp_auth.c

```diff
@@ -0,0 +1,543 @@
+                                    "--client-id=%s",
+                                    idp_auth_ctx->client_id);
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    extra_args[c] = talloc_asprintf(extra_args,
+                                    "--client-secret-stdin");
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    extra_args[c] = talloc_asprintf(extra_args,
+                                    "--token-endpoint=%s",
+                                    idp_auth_ctx->token_endpoint);
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    extra_args[c] = talloc_asprintf(extra_args,
+                                    "--device-auth-endpoint=%s",
+                                    idp_auth_ctx->device_auth_endpoint);
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    extra_args[c] = talloc_asprintf(extra_args,
+                                    "--userinfo-endpoint=%s",
+                                    idp_auth_ctx->userinfo_endpoint);
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    if (idp_auth_ctx->idp_type != NULL
+            && strncasecmp(idp_auth_ctx->idp_type, "keycloak:", 9) == 0) {
+        /* Keycloak is using the 'id' attribute as 'sub' for OIDC */
+        extra_args[c] = talloc_asprintf(extra_args,
+                                        "--user-identifier-attribute=sub");
+    } else {
+        extra_args[c] = talloc_asprintf(extra_args,
+                                        "--user-identifier-attribute=id");
+    }
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    extra_args[c] = talloc_asprintf(extra_args,
+                                    "--scope=%s",
+                                    idp_auth_ctx->scope);
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    chain_id = sss_chain_id_get();
+    extra_args[c] = talloc_asprintf(extra_args,
+                                    "--chain-id=%lu",
+                                    chain_id);
+    if (extra_args[c] == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+    c++;
+
+    if (DEBUG_IS_SET(SSSDBG_TRACE_LIBS)) {
+        extra_args[c] = talloc_asprintf(extra_args, "--libcurl-debug");
```

Comment:
This can be strdup.

But what is the point in general?
Current debug level will be provided to 'oidc_child' anyway, so it can `DEBUG_IS_SET()` internally, right?

Reply:
Changed to strdup. Since oidc_child is used by FreeIPA as well I would prefer to keep the flexibility with the libcurl debug output. I tight it to `SSSDBB_TRACE_LIBS` here to avoid an additional `sssd.conf` option.